src/map/utils/charutils.cpp

```diff
@@ -3957,14 +3957,32 @@ namespace charutils
             // all members might not be in range
             if (!members.empty())
             {
-                // distribute gil
-                int32 gilPerPerson = static_cast<int32>(gil / members.size());
-                for (auto* PMember : members)
+                // Distribute Gil
+                int32 gilPerPerson    = static_cast<int32>(gil / members.size());
+                int16 gilFinderActive = 0;
+
+                for (auto PMember : members)
+                {
+                    // Check for highest gilfinder tier
+                    if (PMember->getMod(Mod::GILFINDER) > gilFinderActive)
+                    {
+                        gilFinderActive = PMember->getMod(Mod::GILFINDER);
+                    }
+                }
+                for (auto PMember : members)
                 {
-                    // Check for gilfinder
-                    gilPerPerson += gilPerPerson * PMember->getMod(Mod::GILFINDER) / 100;
-                    UpdateItem(PMember, LOC_INVENTORY, 0, gilPerPerson);
-                    PMember->pushPacket(new CMessageBasicPacket(PMember, PMember, gilPerPerson, 0, 565));
+                    // Distributte gilfinder gil
+                    if (gilFinderActive > 0)
+                    {
+                        int32 memberGil = gilPerPerson * (100 + gilFinderActive) / 100;
+                        UpdateItem(PMember, LOC_INVENTORY, 0, memberGil);
+                        PMember->pushPacket(new CMessageBasicPacket(PMember, PMember, memberGil, 0, 565));
+                    }
+                    else
+                    {
+                        UpdateItem(PMember, LOC_INVENTORY, 0, gilPerPerson);
+                        PMember->pushPacket(new CMessageBasicPacket(PMember, PMember, gilPerPerson, 0, 565));
+                    }
```

Comment:
I mean, theres no need to do the math per party member if every party member is supposed to get the same amount.

Reply:
> I mean, theres no need to do the math per party member if every party member is supposed to get the same amount

Yeah, I modified the code now to avoid those unneeded calculations, thanks!